From: Matthew Wilcox <matthew.r.wilcox@intel.com>
Date: Sun, 27 Mar 2011 08:52:06 -0400
Subject: NVMe: Fix warning in free_irq
Git-commit: aba2080f3f1639f9202f1a52993669844abcfb80
References: FATE#313627
Patch-Mainline: v3.7-rc1

We need to clear the affinity mask before calling free_irq()

Reported-by: Shane Michael Matthews <shane.matthews@intel.com>
Signed-off-by: Matthew Wilcox <matthew.r.wilcox@intel.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/block/nvme.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/drivers/block/nvme.c b/drivers/block/nvme.c
index 014a7f6..bcc780a 100644
--- a/drivers/block/nvme.c
+++ b/drivers/block/nvme.c
@@ -781,8 +781,10 @@ static int adapter_delete_sq(struct nvme_dev *dev, u16 sqid)
 static void nvme_free_queue(struct nvme_dev *dev, int qid)
 {
 	struct nvme_queue *nvmeq = dev->queues[qid];
+	int vector = dev->entry[nvmeq->cq_vector].vector;
 
-	free_irq(dev->entry[nvmeq->cq_vector].vector, nvmeq);
+	irq_set_affinity_hint(vector, NULL);
+	free_irq(vector, nvmeq);
 
 	/* Don't tell the adapter to delete the admin queue */
 	if (qid) {
-- 
1.7.4.2

